### ISSUE TITLE
Null pointer when using JunctionTable definition

### ISSUE THREAD

**Issue by Agraphie** [requery/requery#422]
Hi,

I get a null pointer exception (see below) when upserting an entity with a junction table.

```Java
@Entity
public abstract class AbstractPerson implements Parcelable, Persistable {

  @Key
  int id;

  @ManyToMany
  @JunctionTable(type = AbstractPersonPicture.class)
  List<AbstractPersonPicture> pictures;
}
```

```Java
@Entity
public abstract class AbstractPicture implements Persistable {
  @Key
  int id;
  String url;
}
```

```Java
```

```Java
@Entity
public abstract class AbstractPersonPicture implements Persistable {

  @Column(name = "referenced_user")
  @ForeignKey(
      delete = ReferentialAction.CASCADE,
      update = ReferentialAction.CASCADE,
      references = AbstractPicture.class)
  @Key
  int pictureId;

  @ForeignKey(
      delete = ReferentialAction.CASCADE,
      update = ReferentialAction.CASCADE,
      references = AbstractPerson.class)
  @Key
  int personId;

  String additionalField = "additionalField";
}
```

```Java
      Person person = new Person();
      person.setId(1);
      Picture picture = new Picture();
      picture.setId(1);
      person.getPictures().add(picture);
      database.upsert(person).blockingGet();
```

Where `database` is a ReactiveEntityStore.

```Java
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
    at java.lang.Class.isAssignableFrom(Class.java:545)
    at io.requery.sql.EntityWriter.updateAssociation(EntityWriter.java:679)
    at io.requery.sql.EntityWriter.updateAssociations(EntityWriter.java:625)
    at io.requery.sql.EntityWriter.insert(EntityWriter.java:441)
    at io.requery.sql.EntityWriter.upsert(EntityWriter.java:483)
    at io.requery.sql.EntityDataStore.upsert(EntityDataStore.java:280)
    at io.requery.reactivex.WrappedEntityStore$8.call(WrappedEntityStore.java:138)
    at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:35)
    at io.reactivex.Single.subscribe(Single.java:2517)
    at io.reactivex.Single.blockingGet(Single.java:2001)
```

I also tried settig `type` to `PersonPicture.class` instead of  `AbstractPersonPicture.class` but this resulted in the compile error 
```Java
Error:Execution failed for task ':app:compileDebugJavaWithJavac'.
> java.lang.IllegalArgumentException: part '<any?>' is keyword
```

**Comment by npurushe** [comment]
Hi thanks for reporting. It would be nice to be able to use PersonPicture.class in the type, however it can cause some strange issues with the compiler's processing framework as that class itself is being generated in that same processing round, however it may be possible.

**Comment by Agraphie** [comment]
Thanks for fixing this. Would it be also possible to insead of using `int id` in the Junction class, the following definition

```Java
@Entity
public abstract class AbstractPersonPicture implements Persistable {

  @Column(name = "referenced_user")
  @ForeignKey(
      delete = ReferentialAction.CASCADE,
      update = ReferentialAction.CASCADE,
      references = AbstractPicture.class)
  @Key
  @OneToOne
  AbstractPicture pictureId;

  @ForeignKey(
      delete = ReferentialAction.CASCADE,
      update = ReferentialAction.CASCADE,
      references = AbstractPerson.class)
  @Key
  @OneToOne
  AbstractPerson personId;

  String additionalField = "additionalField";
}
```

So far this would throw a mapping exception (afaik?), as one side is mapped with ManyToMany and the other with OneToOne. Or an exception will be thrown if no mapping annotation is found (leaving out OneToOne).

Because if we use the junction table with `int` we loose the ability to just say `personPicture.getPersoon()`, right?

**Comment by npurushe** [comment]
There isn't a OneToOne relationship between PersonPicture.pictureId and PictureId, you can have many of the same PictureIds present in the PersonPicture table.

**Comment by Agraphie** [comment]
Thanks for the reply. The problem I am trying to solve is:
I want to mimic the behaviour of requeries automatic junction table generation, but with having full control/an extra column in the junction table.

So I was trying to have the comfortable ability to be able to call `personPicture.getPerson()` and not having to manually join/select the `Person` via the id in `PersonPicture.class` (suggested in my first post). My previous post was just me trying to achieve that and asking if you know how that could work.

I hope I made my question a bit clearer.

**Comment by npurushe** [comment]
Sure understand you are looking to use an object reference in that junction table instead of the id.

**Comment by Agraphie** [comment]
Yes. But is that possible at the moment with requery?

**Comment by npurushe** [comment]
Currently not, although this just a simple findByKey once you have the id

**Comment by npurushe** [comment]
This fix is in version 1.2.0 thanks
